xbmc/windowing/X11/WinSystemX11GLESContext.cpp

```diff
@@ -0,0 +1,309 @@
+/*
+ *  Copyright (C) 2005-2018 Team Kodi
+ *  This file is part of Kodi - https://kodi.tv
+ *
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *  See LICENSES/README.md for more information.
+ */
+
+
+#include "WinSystemX11GLESContext.h"
+#include "GLContextEGL.h"
+#include "utils/log.h"
+#include "windowing/GraphicContext.h"
+#include "guilib/DispResource.h"
+#include "threads/SingleLock.h"
+#include "Application.h"
+#include "X11DPMSSupport.h"
+
+#include "cores/RetroPlayer/process/X11/RPProcessInfoX11.h"
+#include "cores/RetroPlayer/rendering/VideoRenderers/RPRendererOpenGLES.h"
+#include "cores/VideoPlayer/DVDCodecs/DVDFactoryCodec.h"
+#include "cores/VideoPlayer/Process/X11/ProcessInfoX11.h"
+#include "cores/VideoPlayer/VideoRenderers/LinuxRendererGLES.h"
+#include "cores/VideoPlayer/VideoRenderers/RenderFactory.h"
+
+#include "OptionalsReg.h"
```

Comment:
wrong include order and groups, https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md#7-headers

Reply:
I'm not sure about the include groups, because I don't know much about the structure of Kodi.
Would this be correct?


#include "WinSystemX11GLESContext.h"

#include "Application.h"
#include "GLContextEGL.h"
#include "OptionalsReg.h"
#include "X11DPMSSupport.h"
#include "guilib/DispResource.h"
#include "threads/SingleLock.h"
#include "utils/log.h"
#include "windowing/GraphicContext.h"

#include "cores/RetroPlayer/process/X11/RPProcessInfoX11.h"
#include "cores/RetroPlayer/rendering/VideoRenderers/RPRendererOpenGLES.h"
#include "cores/VideoPlayer/DVDCodecs/DVDFactoryCodec.h"
#include "cores/VideoPlayer/Process/X11/ProcessInfoX11.h"
#include "cores/VideoPlayer/VideoRenderers/LinuxRendererGLES.h"
#include "cores/VideoPlayer/VideoRenderers/RenderFactory.h"

#include "platform/linux/OptionalsReg.h"

Reply:
Correct is the following order.
```
#include "WinSystemX11GLESContext.h"

#include "Application.h"
#include "GLContextEGL.h"
#include "OptionalsReg.h"
#include "X11DPMSSupport.h"
#include "cores/RetroPlayer/process/X11/RPProcessInfoX11.h"
#include "cores/RetroPlayer/rendering/VideoRenderers/RPRendererOpenGLES.h"
#include "cores/VideoPlayer/DVDCodecs/DVDFactoryCodec.h"
#include "cores/VideoPlayer/Process/X11/ProcessInfoX11.h"
#include "cores/VideoPlayer/VideoRenderers/LinuxRendererGLES.h"
#include "cores/VideoPlayer/VideoRenderers/RenderFactory.h"
#include "guilib/DispResource.h"
#include "threads/SingleLock.h"
#include "utils/log.h"
#include "windowing/GraphicContext.h"

#include "platform/linux/OptionalsReg.h"
```